src/allocator.h

```diff
@@ -253,12 +254,42 @@ allocator_print_allocated(allocator *al)
    return al->ops->print_allocated(al);
 }
 
+// Return the address of the extent holding page at address 'addr'
+static inline uint64
+allocator_extent_base_addr(allocator *al, uint64 addr)
+{
+   allocator_config *allocator_cfg = allocator_get_config(al);
+   return allocator_config_extent_base_addr(allocator_cfg, addr);
+}
+
+// Is the 'addr' a valid page address?
 static inline bool
-allocator_page_valid(allocator *al, uint64 addr)
+allocator_valid_page_addr(allocator *al, uint64 addr)
```

Comment:
This deals with validating the `page address`, whereas an existing routine, `allocator_page_valid()` looks at extent's reference count, as well, to establish 'validity'.